### PR TITLE
support/remove xstate onboarding

### PR DESCRIPTION
**Pull request by ThomasLaforge**
- refactor onboarding routing
- refactor onboarding terms routing
- refactor onboarding select device routing
- wip refactoring onboarding select use case routing
- fix onboarding/welcome routing
- refactoring onboarding reducers to typescript
- wip better onboarding origin / abortion management
- fix onboarding sub-routing
- wip tutorial connect device routing
- tutorial connect device routing
- Add deviceModelId in onboarding redux state
- Replace url param 'deviceId' by redux 'deviceModelId'
- Remove unused actions from onboarding redux state
- onResult genuinecheck
- BigTutorial
- fix pair nano path
- getStepperStatus
- wip use FlowStepper
- Complete onboarding setup device
- remove sendEvent
- one Tutorial for all use cases
- update use case path
- remove and comment old Tutorial
- screen id enum
- cleanup Tutorial
- use enums
- screens for steps and status update
- use callbacks
- complete routing

## 🦒 Context (issues, jira)

[LIVE-1880](https://ledgerhq.atlassian.net/browse/LIVE-1880)

## 💻  Description / Demo (image or video)

Replace all onboarding routing and state management done with xstate, to a solution using react-router-dom and redux / React Contexte 

## 🖤  Expectations to reach

PR must pass CI, rebase develop if conflicts. Thanks!

- **on QA**: at least one of these two checkboxes must be checked:
  - [ ] a specific test planned is defined on Jira
  - [*] this PR is covered by automatic UI test
- **on delivery**: at least one of these two checkboxes must be checked: 
  - [ ] Option 1: **no impact**: The changes of this PR have ZERO impact on the userland (invisible for users)
  - [ ] Option 2: **atomic delivery**: the changes is atomic and complete (no partial delivery)

### CODE REVIEW

File: src/renderer/components/Onboarding/index.v3.tsx [hunk 1]
```diff
@@ -243,11 +235,13 @@ export function Onboarding({ onboardingRelaunched }: { onboardingRelaunched: boo
...
   }, []);
 
-  const CurrentScreen = screens[state.value];
-
   return (
     <React.Fragment>
-      <Pedagogy isOpen={state.context.pedagogy} onClose={() => sendEvent("CLOSE_PEDAGOGY_MODAL")} onDone={() => sendEvent("SETUP_NEW_DEVICE")} />
+      <Pedagogy
+        isOpen={state.context.pedagogy}
+        onClose={() => sendEvent("CLOSE_PEDAGOGY_MODAL")}
```
Comment (LFBarreto) [inline src/renderer/components/Onboarding/index.v3.tsx:242]: try and useCallback every function in the template

File: src/renderer/reducers/onboarding.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      name: "setPassword",
+      label: "onboarding.breadcrumb.setPassword",
+      options: {
+        showBreadcrumb: true,
+        relaunchSkip: true,
+      },
+    },
+    {
+      name: "analytics",
+      label: "onboarding.breadcrumb.analytics",
```
Comment (LFBarreto) [inline src/renderer/reducers/onboarding.ts:109]: do we need to keep all info of states within the redux? i would just keep a list of step Ids and match it to the actual data elsewhere in the components that use and read this state. :thinking:

File: src/renderer/components/Onboarding/Screens/Tutorial/machines/connectSetupDevice.v3.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import { assign, Machine } from "xstate";
+import { setStepperStatus } from "./helpers";
+
+export const connectSetupDevice = Machine({
```
Comment (LFBarreto) [inline src/renderer/components/Onboarding/Screens/Tutorial/machines/connectSetupDevice.v3.tsx:4]: Are we keeping a machine for the connection device here?

File: src/renderer/components/Onboarding/Screens/SelectDevice/index.v3.tsx [hunk 4]
```diff
@@ -32,24 +37,24 @@ const TitleText = styled(Text)`
...
     },
-    [sendEvent],
+    [dispatch, history],
   );
 
   return (
     <SelectDeviceContainer>
       <TopRightContainer>
-        <Button small onClick={() => sendEvent("PREV")}>
+        <Button small onClick={() => history.push("/onboarding/terms")}>
```
Comment (LFBarreto) [inline src/renderer/components/Onboarding/Screens/SelectDevice/index.v3.tsx:57]: useCallback :pray:

File: src/renderer/reducers/onboarding.ts (hunk 2 above)
Comment (ThomasLaforge) [reply src/renderer/reducers/onboarding.ts:109]: that's the case or it's just not a reviewed zone now

**Review by LFBarreto — APPROVED** [review]

**Pull request merged**